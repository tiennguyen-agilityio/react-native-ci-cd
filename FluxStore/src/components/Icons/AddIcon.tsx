import {memo} from 'react';
import {TouchableOpacity} from 'react-native';
import Svg, {Path} from 'react-native-svg';

// Interfaces
import {IconProps} from '@/interfaces';

// Stores
import {useThemeStore} from '@/stores';

export const AddIcon = memo(
  ({width = 8, height = 8, color, disabled = false, style, onPress}: IconProps) => {
    const {
      theme: {icon},
    } = useThemeStore();

    return (
      <TouchableOpacity
        testID="add-icon"
        activeOpacity={0.8}
        onPress={onPress}
        disabled={disabled}
        style={style}>
        <Svg width={width} height={height} viewBox="0 0 14 14" fill="none">
          <Path
            d="M13 8H8v5c0 .55-.45 1-1 1s-1-.45-1-1V8H1c-.55 0-1-.45-1-1s.45-1 1-1h5V1c0-.55.45-1 1-1s1 .45 1 1v5h5c.55 0 1 .45 1 1s-.45 1-1 1z"
            fill={color ?? icon.default}
          />
        </Svg>
      </TouchableOpacity>
    );
  },
);
