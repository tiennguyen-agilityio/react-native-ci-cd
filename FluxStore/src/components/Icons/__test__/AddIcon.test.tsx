import {render, fireEvent, screen} from '@testing-library/react-native';

import {AddIcon} from '../AddIcon';

describe('AddIcon', () => {
  const onPress = jest.fn();

  it('renders correctly (matches snapshot)', () => {
    const {toJSON} = render(<AddIcon />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls onPress when touched and not disabled', () => {
    render(<AddIcon onPress={onPress} />);

    fireEvent.press(screen.getByTestId('add-icon'));

    expect(onPress).toHaveBeenCalledTimes(1);
  });

  it('does NOT call onPress when disabled', () => {
    render(<AddIcon onPress={onPress} disabled />);

    fireEvent.press(screen.getByTestId('add-icon'));

    expect(onPress).not.toHaveBeenCalledTimes(0);
  });
});
