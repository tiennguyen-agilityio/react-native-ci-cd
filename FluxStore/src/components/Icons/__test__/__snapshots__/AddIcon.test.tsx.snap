// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddIcon renders correctly (matches snapshot) 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="add-icon"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={8}
    bbWidth={8}
    fill="none"
    focusable={false}
    height={8}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 8,
          "width": 8,
        },
      ]
    }
    vbHeight={14}
    vbWidth={14}
    width={8}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M13 8H8v5c0 .55-.45 1-1 1s-1-.45-1-1V8H1c-.55 0-1-.45-1-1s.45-1 1-1h5V1c0-.55.45-1 1-1s1 .45 1 1v5h5c.55 0 1 .45 1 1s-.45 1-1 1z"
        fill={
          {
            "payload": 2147483648,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
